iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/EnvSecretCheck.java

```diff
@@ -0,0 +1,152 @@
+
+  private static final List<String> ENTITIES = List.of("ACCESS", "AMPLITUDE", "ANSIBLE", "ADMIN", "API",
+    "APP", "AUTH", "CLIENT", "CONFIG", "DATABASE", "DB", "ENCRYPTION", "ENV", "FACEBOOK", "FIREBASE", "FTP", "GIT",
+    "GITHUB", "GITLAB", "HONEYCOMB", "JWT", "KEYCLOAK", "KEYRING", "LDAP", "MAIL", "MASTER", "MARIADB", "MSSQL",
+    "MYSQL", "NPM", "OAUTH", "OAUTH2", "PG", "POSTGRES", "REDIS", "REFRESH", "REPLICATION", "ROOT", "RPC", "SA",
+    "SECRET", "SERVER", "SIGN", "SIGNING", "SLACK", "SVN", "USER", "VNC", "WEBHOOK");
+
+  private static final List<String> SECRETS = List.of("CREDENTIALS", "KEY", "PASS", "PASSPHRASE", "PASSWD", "PASSWORD",
+    "SECRET", "TOKEN");
+
+  private static final List<String> EXCLUSIONS = List.of("ALLOW", "DIR", "EXPIRE", "EXPIRY", "FILE", "ID",
+    "LOCATION", "NAME", "OWNER", "PATH", "URL");
+
+  private static final Pattern UPPERCASE_UNDERSCORE_NAME_PATTERN = Pattern.compile("^[A-Z_]+$");
+  private static final Pattern UPPERCASE_DASH_NAME_PATTERN = Pattern.compile("^[A-Z-]+$");
+
+  private static final Pattern CAMELCASE_NAME_PATTERN = Pattern.compile("^[A-Za-z]+$");
+
+  private static final Pattern CAMELCASE_SPLIT_PATTERN = Pattern.compile("(?<!(^|[A-Z]))(?=[A-Z])|(?<!^)(?=[A-Z][a-z])");
+
+  private static final Pattern URL_PATTERN = Pattern.compile("^(http|ftp)s?://");
+
+  private static final String ROOT_PATH_PATTERN = "^/[a-z]*+($|/)";
+  private static final String RELATIVE_PATH_PATTERN = "^./[a-zA-Z_-]*+($|/)";
+  private static final String EXPANSION_PATH_PATTERN = "^\\$\\{[^}]+}/";
+  private static final String PATH_WITH_EXPANSION_PATTERN = "/.*+\\.[a-z0-9]{2,4}$";
+  private static final Pattern PATH_PATTERN = Pattern.compile("(" + ROOT_PATH_PATTERN + "|" + RELATIVE_PATH_PATTERN
+    + "|" + EXPANSION_PATH_PATTERN + "|" + PATH_WITH_EXPANSION_PATTERN + ")");
+  private static final Pattern EXPANSION_DEFAULT_VALUE_PATTERN = Pattern.compile("-(\"[^\"]+\"|[^-]+)$");
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(EnvTree.class, (ctx, instruction) -> instruction
+      .variableAssignments().forEach(envVarAssignment -> checkEnvVariableAssignment(ctx, envVarAssignment)));
+  }
+
+  private static void checkEnvVariableAssignment(CheckContext ctx, KeyValuePairTree envVarAssignment) {
+    if (isSensitiveName(envVarAssignment.key().value()) && isSensitiveValue(envVarAssignment.value().value())) {
+      ctx.reportIssue(envVarAssignment.key(), MESSAGE);
+    }
+  }
+
+  private static boolean isSensitiveName(String name) {
+    List<String> words = splitEnvVarName(name);
+    return isSecretWordOnly(words) || containsSecretEntityWordCombination(words);
+  }
+
+  private static List<String> splitEnvVarName(String name) {
+    if (UPPERCASE_UNDERSCORE_NAME_PATTERN.matcher(name).matches()) {
+      return toUpperCase(name.split("_"));
+    }
+    if (UPPERCASE_DASH_NAME_PATTERN.matcher(name).matches()) {
+      return toUpperCase(name.split("-"));
+    }
```

Comment:
Because I wanted to be sure to only process pure cases like "ACCESS_TOKEN" and not something like "Access-Token_PATH". Somehow an approach to prevent false positives.